src/helpers/findAll.ts

```diff
@@ -54,9 +56,10 @@ function findAllInternal(
   });
 
   if (
+    isHostElement(root) &&
+    predicate(root) &&
```

Comment:
[nit] Could we put predicates from cheapest to (potentially) most expensive?

1. deepestOnly
2. host element
3. passed `predicate`